### PR TITLE
fix: don't crash richtext field when string is empty

### PR DESCRIPTION
**Pull request by chrisvxd**
## Description

Fixes a crash in RichTextRender when a richtext field contains an empty string, impacting AI generation.

ProseMirror rejects empty text nodes, so programmatically setting a richtext field to `""` (or initializing it as empty) would throw `RangeError: Empty text nodes are not allowed`, preventing the component from rendering and blocking subsequent updates.

## Changes made

- Added an early return in `RichTextRender` for empty strings, returning an empty doc (`{ type: "doc", content: [] }`) instead of an invalid text node.
- Added focused test coverage for programmatic richtext field updates via `dispatch`, including the empty string edge case and plain-text transition.
- Removed the broader multi-field dispatch test based on review feedback to keep this suite scoped to richtext behavior.

## How to test

Create a component with a richtext field initialized to `""`, then programmatically update it via dispatch. The updated content should render correctly:

```ts
// Initial data with empty richtext
const data = {
  content: [{ type: "Block", props: { id: "1", body: "" } }],
};

// Later, dispatch a replace to set the value
dispatch({
  type: "replace",
  data: { type: "Block", props: { id: "1", body: "<p>Hello</p>" } },
  destinationIndex: 0,
  destinationZone: "default-zone",
});
```

### CODE REVIEW

File: packages/core/components/Puck/__tests__/richtext.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+    });
+    await flush();
+
+    await waitFor(() => {
+      expect(screen.getByText("Updated content")).toBeInTheDocument();
+    });
+    expect(screen.queryByText("Hello world")).not.toBeInTheDocument();
+  });
+
+  it("updates richtext when only the richtext field changes on a component with multiple fields", async () => {
```
Comment (FedericoBonel) [inline packages/core/components/Puck/__tests__/richtext.spec.tsx]: This seems more of a dispatch test than a rich text one, also is this one needed?

File: packages/core/components/Puck/__tests__/richtext.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+type PuckInternal = {
+  appStore: AppStoreApi;
+};
+
+const getInternal = () => {
+  return (window as any).__PUCK_INTERNAL_DO_NOT_USE as PuckInternal;
+};
+
+const flush = () => act(async () => {});
```
Comment (FedericoBonel) [inline packages/core/components/Puck/__tests__/richtext.spec.tsx:72]: Not now, but we should potentially pull these setups into a shared test utils

**Review by FedericoBonel — APPROVED** [review]

File: packages/core/components/Puck/__tests__/richtext.spec.tsx (hunk 1 above)
Comment (chrisvxd) [reply packages/core/components/Puck/__tests__/richtext.spec.tsx]: @copilot remove this
